### ISSUE TITLE
Enhanced dependencies for cross-serial vs distributive semantics of relations

### ISSUE THREAD

**Issue by danielhers** [UniversalDependencies/docs#876]
Enhanced UD does not seem to handle "respectively" as we would expect. 
[texput.pdf](https://github.com/UniversalDependencies/UD_English-EWT/files/8719399/texput.pdf)
Here is an example from the reviews part. The enhanced graph is below the sentence. Note how all appos edges attach to the first "roll".

**Comment by nschneid** [comment]
Oh boy...I gather the "correct" treatment of "respectively" is an old theoretical debate. But I agree it would make more sense to use `appos` links between the respective pairs (what is shown here is probably due to automatic application of a general rule for coordination). Feel free to submit a PR to the dev branch.

**Comment by amir-zeldes** [comment]
I'm not sure if that's really a syntactic relation - as a parenthetical, it would at best be `parataxis`. I can't invert them and keep the verb:

  * #We tried $15 (the crab roll) ...

Semantically of course respectively has that interpretation, but syntactically it's possibly the right thing to say that the two structures are connected (I'd say via parataxis, not appos), but not cross-wise.

**Comment by nschneid** [comment]
Yeah neither `parataxis` nor `appos` really conveys that a parenthetical attribute of an item is being given....I think if you look through EWT you'll find a ton of inconsistency there.

In terms of the tree structure, let's set aside the parenthetical issue and consider

- Alice, Bob, and Charlie will meet David, Eva, and Frank respectively

Clearly, the basic analysis is a coordinated subject and a coordinated object. With EUD there would normally be propagation of relations across coordination. I think the correct EUD analysis would have three "meet" predicates, 2 of them copy nodes, so it is nsubj(meet, Alice), obj(meet, David), nsubj(meet', Bob), obj(meet', Eva), etc. 

The alternative would be a fully distributive representation where the EUD tree has one "meet" with 3 subjects and 3 objects. I'm not sure why we would want that.

TL;DR I do think the "respectively" interpretation as distinct from the fully distributive representation is relevant to how the coordination dependencies are propagated in EUD.

**Comment by danielhers** [comment]
Thank you both for the interesting debate. I thought there must be agreement in how this _should_ be annotated but apparently there isn't :)
I tend to agree with Nathan that EUD should reflect the cross-wise semantics rather than stick to a strictly "syntactic" approach. It would be much more useful this way.
So before the PR we must resolve these two questions:

1. What relation to use for parenthetical attributes?
2. What enhanced head to select for a cross-wise interpretation as opposed to a fully distributive one?

**Comment by amir-zeldes** [comment]
I like semantics as much as the next person (probably more than a majority of next persons :) but I don't think we have thought this through here. Regardless of the distributive semantics, "Alice, Bob, and Charlie" is just a coordinate NP. If you think there should be cross-dependencies with empty nodes here, what do you think about:

  * Bob and Charlie love each other

Should there be two "love" nodes, one with edeps to Bob as the subject and Charlie as the object, and the other the other way around? If so, then I think edeps are currently doing something different than what you are expecting for a lot of phenomena involving scope, distribution and more. If not, why should "respectively" be different?

**Comment by nschneid** [comment]
I did not mean to suggest that EUD should represent semantics over syntax in general, but I thought the aim of having coordination-based propagation annotated in a separate layer was that it was not always possible to infer these by rule. Maybe I was mistaken, though:

The [EUD guidelines](https://universaldependencies.org/u/overview/enhanced-syntax.html#propagation-of-incoming-dependencies-to-conjuncts) say

> However, as the distinction between distributive and collective readings is often context-dependent, we take the simplest approach and always attach all conjuncts to the predicate.

So does this mean the propagation of dependencies due to coordination is fully deterministic? If so should the validator ensure that it matches the Basic dependencies + the rest of the EUD graph?

**Comment by amir-zeldes** [comment]
I think this might be a good issue to float to the core group, since this is really about the scope of edeps in general. I feel like we shouldn't add these to specific constructions like 'respectively', no matter how tempting/fixable it looks, when there are large areas of grammar cross-linguistically (all reciprocals and reflexives to name just two) that would be affected or should behave consistently with this decision.

**Comment by sylvainkahane** [comment]
I agree with @amir-zeldes. _respectively_ is optional and likely to be absent in non formal utterance and sometimes it is impossible even if semantically there is a "cross-serial" interpretation:

```
Peter and John respectively married Ann and Mary
Peter and John married Ann and Mary
Peter and John married the Smith sisters
*Peter and John respectively married the Smith sisters
```

**Comment by danielhers** [comment]
@dan-zeman just wondering, is this incorporated into the EUD guidelines or closed as "won't fix"?

**Comment by dan-zeman** [comment]
Sorry if I was too eager in closing the issue – I am trying to clean up the issue tracker around every release. This one had no activity for half a year, and the last posts seemed to be just a discussion about the best analysis of _respectively_, without any obvious consequences for the guidelines. However, I now see that @amir-zeldes actually suggested to discuss it in the core group, so I'm reopening it and setting a new milestone.
